heed/src/txn.rs

```diff
@@ -109,15 +125,35 @@ impl<'e, T> RoTxn<'e, T> {
     pub fn commit(mut self) -> Result<()> {
         // Asserts that the transaction hasn't been already
         // committed/aborter and ensure we cannot use it twice.
-        let mut txn = self.txn.take().unwrap();
+        let mut txn = self.inner.txn.take().unwrap();
         let result = unsafe { mdb_result(ffi::mdb_txn_commit(txn.as_mut())) };
         result.map_err(Into::into)
     }
 }
 
+impl<'a> Deref for RoTxn<'a, WithTls> {
+    type Target = RoTxn<'a, AnyTls>;
+
+    fn deref(&self) -> &Self::Target {
+        // SAFETY: OK because repr(transparent) means RoTxn<T> always has the same layout
+        // as RoTxnInner.
+        unsafe { std::mem::transmute(self) }
+    }
+}
+
+impl<'a> Deref for RoTxn<'a, WithoutTls> {
+    type Target = RoTxn<'a, AnyTls>;
+
+    fn deref(&self) -> &Self::Target {
+        // SAFETY: OK because repr(transparent) means RoTxn<T> always has the same layout
+        // as RoTxnInner.
+        unsafe { std::mem::transmute(self) }
+    }
+}
+
 impl<T> Drop for RoTxn<'_, T> {
     fn drop(&mut self) {
-        if let Some(mut txn) = self.txn.take() {
+        if let Some(mut txn) = self.inner.txn.take() {
```

Comment:
Out of scope but should I really wrap this `NonNull` into an `Option` when I am sure that this struct cannot be dropped two times, it's impossible?